components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/context/CloudControllerContext.java

```diff
@@ -161,6 +158,20 @@
 	 */
 	private Map<String, NetworkPartition> networkPartitionIDToNetworkPartitionMap;
 
+
+    /**
+     * Key - cartridge id (cartridge type is used as an unique identifier)
```

Comment:
IMO it would be better to use Cartridge Type instead of Cartridge ID as we are already using Cartridge Type as the key of the cartridge.
Thanks